### ISSUE TITLE
Castling does not work correctly

### ISSUE THREAD

**Issue by EarlRussellJr** [oakmac/chessboardjs#133]
I play this opening set of moves on chessboard.js and the castle move only moves the King and not the Rook.

Ruy Lopez Open
1.e4 e5 2.f3 c6 3.b5 a6 4.a4 f6 5.O-O xe4

Chess.js translates to:
 1. e2-e4 e7-e5
 2. g1-f3 b8-c6 
3. f1-b5 a7-a6
 4. b5-a4 g8-f6
 5. e1-g1 f6-e4

Upon move 5 for white, o-o, the actual move should be e1-g1, h1-f1 for white.  
The board moves the King but does not move the White Rook.

Chessboard.js seems to be intelligent enough to allow the King to move two squares as a legal move.
Since this action can occur only once in a game then why doesn't the code move the Rook as well, as in this instance that is a legal chess move?  I say this because the web site indicates you can control legal moves on the board.

Code

In widget.move = function() {

I added a conditional statement to look for the castle move:

    if (tmp[0] === 'e1' && tmp[1] === 'g1') {
        kingCastle = true;
    }
    
moves[tmp[0]] = tmp[1];

Then I attempted in interject the code as such:

    	//if o-o repeat move for KR e1,g1    
	//if o-o-o repeat move for QR
  
if (kingCastle) {
      moves['e1'] = 'g1';  //move King
      newPos = calculatePositionFromMoves(CURRENT_POSITION, moves);
      widget.position(newPos, useAnimation);

      moves['h1'] = 'f1';  //move Rook
      newPos = calculatePositionFromMoves(CURRENT_POSITION, moves);
      widget.position(newPos, useAnimation);
  }
  else {
      // calculate position from moves      
	var newPos = calculatePositionFromMoves(CURRENT_POSITION, moves);
      
	// update the board      
	widget.position(newPos, useAnimation);  
}
  // return the new position object  
return newPos;

I tried a couple of different combinations but had no success.  In the debugger it says wR is undefined, so it did not accept my attempt to move the rook either before or after the king move.

Is there a fix for the castle issue?  If not could you make some suggestions to try?

If there is no fix but I am able to experiment and get one to work I'll submit it back to you for approval.

Thank you.
Earl Russell

BTW - I have not tested o-o-o but I assume it has the same issue.

**Comment by oakmac** [comment]
chessboard.js is a "dumb board" that does not understand the rules of the game, like how pieces move, whether castling is allowed, etc. This logic is captured in [chess.js](https://github.com/jhlywa/chess.js), which works well with chessboard.js.

**Comment by EarlRussellJr** [comment]
Thank you for the quick response.
If what you say is correct, and I assume it is, then how do you allow two moves in sequence for one color so that a castle operation can be completed?  Your code only allows one set of coordinates per move as far as I can see by looking at the JavaScript source, and after that the next move goes to the other color..  If you allowed o-o as a legal move that would solve the issue, but it appears that is not so.  I think you have had this 'complaint' before but I could find no solution either here or on the chess.js site.  Do you have any suggestions?

**Comment by oakmac** [comment]
You can pass multiple moves to the `.move()` method as seen in [Example 3004](http://chessboardjs.com/examples#3004)

```js
// white kingside castle
myBoard.move('e1-g1', 'h1-f1');
```

**Comment by EarlRussellJr** [comment]
Thank you!  I will remove my test code and give that a try tomorrow.  If this works, and I assume it will, I will notify and discuss the issue with chess.js.  I use their code to convert a PGN file to the format needed for your board, so it looks like I will need to add conditional code for castle options.  Should be interesting.  So far your board has been working nicely and I would like to say 'job well done'!  
Again, thank you for your quick responses.

**Comment by oakmac** [comment]
Thank you for the kind words. Glad you like the project :)

**Comment by EarlRussellJr** [comment]
One comment/recommendation I would like to suggest that you add to your documentation.  I am using your board in an ASP.NET MVC 5 project and I placed your images within my images folder.  I had to go find the path to the images and make one change.  It would be nice if there were a comment somewhere and maybe indicate what a change needs to be made for whatever configuration a developer may decide.  Or make it a global variable at the top of the .js file.  My solutions was 
  if (cfg.hasOwnProperty('pieceTheme') !== true ||
      (typeof cfg.pieceTheme !== 'string' &&
       typeof cfg.pieceTheme !== 'function')) {
    cfg.pieceTheme = '../Images/chesspieces/wikipedia/{piece}.png';
  }
and all is well with the images.
Thanks,
Earl Russell
